keel-core/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/NpmVersioningStrategy.kt

```diff
@@ -0,0 +1,20 @@
+package com.netflix.spinnaker.keel.artifacts
+
+import com.netflix.spinnaker.keel.api.artifacts.VersioningStrategy
+
+/**
+ * A [VersioningStrategy] that for NPM packages.
+ */
+object NpmVersioningStrategy : VersioningStrategy {
+  override val type: String = "npm"
```

Comment:
The fact that this has a different name than the version strategy we were using previously shouldn't have any impact as it's not present in delivery configs or persisted to the database.

Reply:
Are you sure about that? It appears in the JSON schema. If it's not in the delivery config that's a bug

Reply:
I was surprised by this initially when I changed the name from `deb` to `netflix-semver` a couple months back and it didn't break, but I found the reason is that the versioning strategies for debian and npm are hard-coded as defaults and we don't really support anything else for those artifacts (see `supportedVersioningStrategy` below), so if a user would specify something different, _then_ things would break. I understand this is not "correct" per the schema, but would prefer to address that separately if you think it's worth the trouble.

Debian: https://github.com/spinnaker/keel/pull/1510/files#diff-ed217978125ec043314d9063f798b04fR18

NPM: https://github.com/spinnaker/keel/pull/1510/files#diff-dbe8bb0ebf9249a6504795a1236a1d61R16

Reply:
@robfletcher I'm merging this so I can test it with live data, but feel free to comment if you disagree with the approach.

Reply:
P.S. About the fact that it appears in the schema, it should still be marked optional since it has a default, correct? I don't expect any users to be specifying the versioning strategy for Debian or NPM, and given that the schema just went live (meaning no one would have seen the old discriminator), I hope this will be OK.